README.md

```diff
@@ -2,11 +2,876 @@
+  }
+}
+```
+
+4. Create a RestController for FirstService
+5. Create a span to wrap the FirstServiceController
+
+```java
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+import HttpUtils;
+
+@RestController
+@RequestMapping(value = "/message")
+public class FirstServiceController {
+  @Autowired
+  private Tracer tracer;
+
+  @Autowired
+  HttpUtils httpUtils;
+
+  private static String SS_URL = "http://localhost:8081/time";
+
+  @GetMapping
+  public String firstTracedMethod() {
+    Span span = tracer.spanBuilder("message").startSpan();
+    span.addEvent("Controller Entered");
+    span.setAttribute("what.are.you", "Je suis attribute");
+
+    try (Scope scope = tracer.withSpan(span)) {
+      return "Second Service says: " + httpUtils.callEndpoint(SS_URL);
+    } catch (Exception e) {
+      span.setAttribute("error", e.toString());
+      span.setAttribute("error", true);
+      return "ERROR: I can't tell the time";
+    } finally {
+      span.end();
+    }
+  }
+}
+```
+
+6. Configure `HttpUtils.callEndpoint` to inject span context into request. This is key to propagate the trace to the SecondService
+
+HttpUtils is a helper class that injects the current span context into outgoing requests. This involves adding the tracer id and the trace-state to a request header. For this example, I used `RestTemplate` to send requests from `FirstService` to `SecondService`. A similar approach can be used with popular Java Web Clients such as [okhttp](https://square.github.io/okhttp/) and [apache http client](https://www.tutorialspoint.com/apache_httpclient/apache_httpclient_quick_guide.htm). The key to this implementation is to override the put method in `HttpTextFormat.Setter<?>` to handle your request format. `HttpTextFormat.inject` will use this setter to set `traceparent` and `tracestate` headers in your requests. These values will be used to propagate your span context to external services.
+
+```java
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.Tracer;
+
+@Component
+public class HttpUtils {
+
+  @Autowired
+  private Tracer tracer;
+
+  private HttpTextFormat<SpanContext> textFormat;
+  private HttpTextFormat.Setter<HttpHeaders> setter;
```

Comment:
[Note] Finals where applicable on class and field. 

Within Google3, this is usually a major readability nit and all the static analyzers would go crazy if they spot a field that hasn't been marked.

I believe it's also in the standard style guide.